### ISSUE TITLE
Android custom network adapter not working

### ISSUE THREAD

**Issue by pm-preety-narvekar** [AppLovin/AppLovin-MAX-SDK-Android#212]
Hello,

We have developed a custom network adapter for banner and interstitial ads only based on the [documentation](https://dash.applovin.com/documentation/mediation/auctions/custom-adapter-android), and followed the steps as described [here](https://dash.applovin.com/documentation/mediation/android/mediation-setup/custom-sdk) in order to define and enable the new adapter in your Dashboard.

The issue is that `initialize` never gets invoked. Neither other overwritten methods such as `getSdkVersion`, `getAdapterVersion` etc. I have carefully named my class as instructed by your documentation and placed them on app side in the correct package and also cross-checked them with your supported mediation adapter implementation found in this repository. Also, I have initialized the SDK, added SDK key in menifest file, used correct applicationId in the app similar to the app in the portal setup.

Additionally, I can see a reference to our custom network in the console logs when verbose logging is enabled. but also can see below error:
`E/AppLovinSdk: [AdUnitWaterfall] Failed to retrieve network info for adapter class: <MyAdapterName_withPackage>`
But I can also see MAX ad being rendered.

Do you have any suggestions on how to proceed?

Note: We are facing similar issue in iOS and have raised the issue at https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/issues/68

**Comment by joeapplovin** [comment]
Hi @pm-preety-narvekar , that error message you're seeing is from the mediation debugger, which is expected at the moment.
Would it be possible for you to provide a sample project of your set up so I can debug why it isn't working for you?

**Comment by pm-preety-narvekar** [comment]
Thanks @joeapplovin for your response. Please check this attached app:
[MAXCustomAdaptersSample.zip](https://github.com/AppLovin/AppLovin-MAX-SDK-Android/files/8182645/MAXCustomAdaptersSample.zip)

Please note, this app has only banner integration and I have just added the adapters with the required methods (with logs) as per your guide. Implementation of those methods are not yet added.

**Comment by pm-preety-narvekar** [comment]
Hi @joeapplovin,

Can you guide me if you got any clue about this?

**Comment by joeapplovin** [comment]
Hi @pm-preety-narvekar - would it be possible for you to try to run either of these [custom adapter](https://github.com/AppLovin/AppLovin-MAX-SDK-Android/tree/master/CustomAdapters) in your project just to confirm your waterfall is configured correctly to return custom networks in the UI?

**Comment by pm-preety-narvekar** [comment]
Hi @joeapplovin,
It is working fine on device. Thanks for all your support.

**Comment by joeapplovin** [comment]
Hi Ritesh,

Correct, please do not enable test mode as that will return a different
waterfall and won’t contain the adapter you’re trying to use.

**Comment by pm-preety-narvekar** [comment]
Hi @riteshghodrao ,

I was trying on an emulator, so it was not working. I checked on the device and I was able to get control in all the methods in my adapter. I tried the below things:

1. Test mode was not enabled from the App side.
2. In setup, I enabled the bid floor and set it to exact same value as the CPM of my custom network (10$ for testing purpose).
3. Recheck the app package name, ad unit, SDK key.
4. Try on the real device.
